### PR TITLE
Spice openff

### PR DESCRIPTION
**Pull request by chrisiacovella**
## Description
This PR does two main things. 
 (1) it provides curation routines for the spice dataset at the openff level of theory  fetching from qcarchive
(2) it makes array shapes in the generated hdf5 files consistent (see Issue: #29 ) 
(3) This also addresses breaking changes to the zenodo api (see Issue #31 ), fixing routines we use and removing ones I wrote that we ultimately did not end up using. 

## Todos
Notable points that this PR has either accomplished or will accomplish.
  - [x] As discussed in issue #29 , need to modify the attribute for each field to denote if a property is per-atom or per-molecule, in addition to the "series" info (that says if the property contains information that varies per conformer).  This is necessary because simply looking at the shape of an array is in itself insufficient.  This should not be hard, but need to decide on the most appropriate way to define it (and annotate in the code).
  - [x] Add in checksum verification for figshare/zenodo downloader

## Questions
- [x] For Spice openff, do we want the hdf5 file to include the dispersion correction energy/forces separate or only have a single corrected value?  Is there any reason for these values to be separate? Consensus is to only include dispersion corrected energies and forces. 

## Status
- [x] Ready to go

### CODE REVIEW

**Comment by chrisiacovella** [comment]
I think this is now ready to go.  Some of the data loading is a bit hacky, as some of the changes to the shape of the underlying data arrays was done purposefully to make it easier for the diagonal batching that is being worked on.  (e.g., I need to reshape the atomic numbers of shape [n,1], i.e., [[12],[1], [1], [1], [1]] to a [n] array);  the pad_to_max_length function was updated then to operate on np arrays rather than lists, since I think all of our underlying data should be of the form of an array.  Similarly, energy is stored is an [m,1] array.  For QM9, since m =1, this is just a single float value. To keep things compatible at this moment, this single float value is extracted and appended to the array (adding an array was failing; not sure how much effort to put into this considering we will be changing the approach).

**Comment by chrisiacovella** [comment]
@ArnNag Can you take a look?

File: modelforge/dataset/utils.py [hunk 1]
```diff
@@ -237,10 +237,10 @@ def pad_to_max_length(data: List[np.ndarray]) -> List[np.ndarray]:
     List[np.ndarray]
         List of padded arrays.
     """
-    max_length = max(len(arr) for arr in data)
+    max_length = max(arr.shape[0] for arr in data)
 
     return [
-        np.pad(arr, (0, max_length - len(arr)), "constant", constant_values=0)
```
Comment (wiederm) [inline modelforge/dataset/utils.py:243]: nice!

File: modelforge/tests/test_remote.py [hunk 2]
```diff
@@ -5,8 +5,6 @@
 
 from modelforge.utils.remote import *
 
-IN_GITHUB_ACTIONS = os.getenv("GITHUB_ACTIONS") == "true"
-
 
 @pytest.fixture(scope="session")
 def prep_temp_dir(tmp_path_factory):
```
Comment (wiederm) [inline modelforge/tests/test_remote.py:10]: might make sense to put this in a `conftest.py` file

**Review by wiederm — APPROVED** [review]
This is great! I don't see any blocker

File: modelforge/tests/test_remote.py (hunk 2 above)
Comment (chrisiacovella) [reply modelforge/tests/test_remote.py:10]: Ah, so this just got left in accidentally, as I was testing some different ways to mark tests for local vs. CI.

**Pull request merged**